kiwi/xml_description.py

```diff
@@ -108,10 +119,21 @@ def __parse(self):
 
     def _xsltproc(self):
         """
-        Transform XML description with the Kiwi rules.
+        Apply XSLT style sheet rules to the XML data
 
-        :return:
+        The result of the XSLT processing is stored in a named
+        temporary file and used for further schema validation
+        and parsing into the data structure classes
         """
-        transform = etree.XSLT(etree.parse(Defaults.get_xsl_stylesheet_file()))
+        xslt_transform = etree.XSLT(
+            etree.parse(Defaults.get_xsl_stylesheet_file())
+        )
+
+        xml_source = self.description
+        if self.xml_content:
+            xml_source = BytesIO(bytes(self.xml_content))
+
```

Comment:
I always try to avoid re-assigning and thus I would write it:
```python
if self.xml_content:
    xml_source = BytesIO(bytes(self.xml_content))
else:
    xml_source = self.description
```
But this is looks like a Pascal in my student's ages. Therefore use ternary operator:
```python
xml_source = BytesIO(bytes(self.xml_content)) if self.xml_content else self.description
```
Or 1 char shorter:
```python
xml_source = self.xml_content and BytesIO(bytes(self.xml_content)) or self.description
```